### PR TITLE
BUG: Collate gathered DeepMIL outputs on CPU

### PR DESCRIPTION
**Pull request by dccastro**
DeepMIL outputs are currently "`all_gather`ed" across GPUs, but each tensor remains on the respective device. This PR addresses potential issues when attempting to process collections of output tensors on different devices by moving them to CPU.

### CODE REVIEW

File: hi-ml-histopathology/testhisto/testhisto/utils/test_output_utils.py [hunk 1]
```diff
@@ -214,15 +217,34 @@ def test_gather_results_distributed() -> None:
...
+    _test_collate_results(epoch_results, total_num_samples=num_batches * batch_size)
+
+
+@pytest.mark.skipif(torch.cuda.device_count() < 2, reason="Not enough GPUs available")
+@pytest.mark.gpu
+def test_collate_results_multigpu() -> None:
+    num_batches = 5
+    batch_size = 3
+    epoch_results = _create_epoch_results(batch_size, num_batches, rank=0, device='cuda:0') \
+        + _create_epoch_results(batch_size, num_batches, rank=1, device='cuda:1')
```
Comment (fepegar) [inline hi-ml-histopathology/testhisto/testhisto/utils/test_output_utils.py:249]: Can we do this in three lines to avoid `\`?

**Review by fepegar — CHANGES_REQUESTED** [review]

**Review by ant0nsc — APPROVED** [review]

**Review by harshita-s — APPROVED** [review]

File: hi-ml-histopathology/testhisto/testhisto/utils/test_output_utils.py [hunk 2]
```diff
@@ -168,12 +168,15 @@ def _create_batch_results(batch_idx: int, batch_size: int, num_batches: int, ran
...
                              for bag_size in bag_sizes],
         ResultsKey.BAG_ATTN: [torch.rand(1, bag_size, device=device) for bag_size in bag_sizes],
+        ResultsKey.TRUE_LABEL: torch.randint(2, size=(batch_size,), device=device),
         ResultsKey.LOSS: torch.randn(1, device=device),
     }
     for key, values in results.items():
         if key is ResultsKey.LOSS:
             continue  # loss is a scalar
         assert len(values) == batch_size
+        if key is ResultsKey.TRUE_LABEL:
```
[the remark this answers is not in the document]
Comment (dccastro) [reply hi-ml-histopathology/testhisto/testhisto/utils/test_output_utils.py:178]: For basic types, sure. But here `ResultsKey` is an `Enum` (of which `key` is a member), so [should be compared by identity](https://docs.python.org/3/library/enum.html#comparisons).

[the remark this answers is not in the document]
Comment (fepegar) [reply hi-ml-histopathology/testhisto/testhisto/utils/test_output_utils.py:178]: Interesting, I didn't know that one. Thanks!

File: hi-ml-histopathology/testhisto/testhisto/utils/test_output_utils.py (hunk 1 above)
Comment (dccastro) [reply hi-ml-histopathology/testhisto/testhisto/utils/test_output_utils.py:249]: Is it bad enough to block the PR? 🙃

**Review by fepegar — APPROVED** [review]

**Pull request merged**